{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "    --input-path Data/raw_data/ \\\n",
    "    --input-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "    --output-path Data/demux-paired-end.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime demux summarize \\\n",
    "    --i-data Data/demux-paired-end.qza \\\n",
    "    --o-visualization Results/demux-paired-end.qzv "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Use cutadapt to remove primers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime cutadapt trim-paired \\\n",
    "    --i-demultiplexed-sequences Data/demux-paired-end.qza \\\n",
    "    --o-trimmed-sequences Data/trimmed-paired-end.qza \\\n",
    "    --p-cores 18 \\\n",
    "    --p-front-f AGATGTGTATAAGAGACAGCCTACGGGNGGCNGCAG \\\n",
    "    --p-front-r GACTACNNGGGTATCTAATCC \\\n",
    "    --p-match-adapter-wildcards\n",
    "# Visualization\n",
    "!mkdir Results\n",
    "!qiime demux summarize \\\n",
    "    --i-data Data/trimmed-paired-end.qza \\\n",
    "    --o-visualization Results/trimmed-paired-end.qzv "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Denoising with DADA2\n",
    "Note. We modified R script for Dada2 (file \"anaconda3/envs/qiime2-2019.4/bin/run_dada_paired.R\") by providing “mergers[[j]] <- mergePairs(ddF, drpF, ddR, drpR, minOverlap=4)” to allow change minimum overlap to 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir Data/tables\n",
    "!qiime dada2 denoise-paired \\\n",
    "    --i-demultiplexed-seqs Data/trimmed-paired-end.qza \\\n",
    "    --p-trim-left-f 2 \\\n",
    "    --p-trim-left-r 2 \\\n",
    "    --p-trunc-len-f 267 \\\n",
    "    --p-trunc-len-r 238 \\\n",
    "    --p-n-threads 18 \\\n",
    "    --o-table Data/tables/table.qza \\\n",
    "    --o-representative-sequences Data/rep-seqs.qza \\\n",
    "    --o-denoising-stats Data/denoising-stats.qza\n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "    --i-data Data/rep-seqs.qza \\\n",
    "    --o-visualization Results/rep-seqs.qzv\n",
    "\n",
    "!qiime metadata tabulate \\\n",
    "    --m-input-file Data/denoising-stats.qza \\\n",
    "    --o-visualization Results/denoising-stats.qzv\n",
    "\n",
    "!mkdir -p Results/tables\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/tables/table.qza \\\n",
    "    --o-visualization Results/tables/table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv\n",
    "\n",
    "!rm Data/demux-paired-end.qza Data/trimmed-paired-end.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxonomic analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!qiime feature-classifier classify-sklearn \\\n",
    "    --i-classifier training-feature-classifiers/classifier.qza \\\n",
    "    --i-reads Data/rep-seqs.qza \\\n",
    "    --o-classification Data/taxonomy.qza\n",
    "    #--p-reads-per-batch 20000 \\\n",
    "\n",
    "!qiime tools export \\\n",
    "    --input-path Data/taxonomy.qza \\\n",
    "    --output-path Data/taxonomy-with-spaces\n",
    "!qiime metadata tabulate \\\n",
    "    --m-input-file Data/taxonomy-with-spaces/taxonomy.tsv  \\\n",
    "    --o-visualization Data/taxonomy-as-metadata.qzv\n",
    "!qiime tools export \\\n",
    "    --input-path Data/taxonomy-as-metadata.qzv \\\n",
    "    --output-path Data/taxonomy-as-metadata\n",
    "!qiime tools import \\\n",
    "    --type 'FeatureData[Taxonomy]' \\\n",
    "    --input-path Data/taxonomy-as-metadata/metadata.tsv \\\n",
    "    --output-path Data/taxonomy-without-spaces.qza\n",
    "!qiime metadata tabulate \\\n",
    "    --m-input-file Data/taxonomy-without-spaces.qza \\\n",
    "    --o-visualization Results/taxonomy-without-spaces.qzv\n",
    "!rm -r Data/taxonomy-with-spaces Data/taxonomy-as-metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BIOM editing to add taxa to ASVs hashes in BIOM table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/timyerg/anaconda3/envs/qiime2-2020.2\n",
      "\n",
      "  added / updated specs:\n",
      "    - biopython\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    biopython-1.76             |   py36h7b6447c_0         2.6 MB  anaconda\n",
      "    certifi-2020.4.5.1         |           py36_0         159 KB  anaconda\n",
      "    openssl-1.1.1g             |       h7b6447c_0         3.8 MB  anaconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         6.6 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  biopython          anaconda/linux-64::biopython-1.76-py36h7b6447c_0\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2020.4.5~ --> anaconda::ca-certificates-2020.1.1-0\n",
      "  certifi            conda-forge::certifi-2020.4.5.1-py36h~ --> anaconda::certifi-2020.4.5.1-py36_0\n",
      "  openssl            conda-forge::openssl-1.1.1g-h516909a_0 --> anaconda::openssl-1.1.1g-h7b6447c_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "openssl-1.1.1g       | 3.8 MB    | ##################################### | 100% \n",
      "certifi-2020.4.5.1   | 159 KB    | ##################################### | 100% \n",
      "biopython-1.76       | 2.6 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n"
     ]
    }
   ],
   "source": [
    "!conda install -y -c anaconda biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mExported Data/tables/table.qza as BIOMV210DirFmt to directory Biom/\u001b[0m\n",
      "\u001b[32mExported Data/taxonomy-without-spaces.qza as TSVTaxonomyDirectoryFormat to directory Taxa\u001b[0m\n",
      "\u001b[32mExported Data/rep-seqs.qza as DNASequencesDirectoryFormat to directory Rep-seqs/\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#input\n",
    "table    = 'Data/tables/table.qza'\n",
    "taxa     = 'Data/taxonomy-without-spaces.qza'\n",
    "rep_seq  = 'Data/rep-seqs.qza'\n",
    "\n",
    "#import\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "#export rep-seqs.qza, table.qza and taxonomy.qza\n",
    "!mkdir Biom Taxa Rep-seqs\n",
    "\n",
    "!qiime tools export \\\n",
    "    --input-path $table \\\n",
    "    --output-path Biom/\n",
    "!biom convert -i Biom/feature-table.biom -o Biom/feature-table.tsv --to-tsv\n",
    "\n",
    "!qiime tools export \\\n",
    "    --input-path $taxa \\\n",
    "    --output-path Taxa\n",
    "\n",
    "!qiime tools export \\\n",
    "    --input-path $rep_seq \\\n",
    "    --output-path Rep-seqs/\n",
    "\n",
    "#replacing hashes with combination of taxonomy and beginings of the hashes\n",
    "taxa_df = pd.read_csv('Taxa/taxonomy.tsv', sep='\\t', skiprows=[1])\n",
    "biom_df = pd.read_csv('Biom/feature-table.tsv', sep='\\t', skiprows=1)\n",
    "\n",
    "taxa_df['Combo'] = pd.DataFrame(taxa_df['Taxon'].astype(str).str.replace(';__', '').str.replace('[', '').str.replace(']', '')\n",
    "                                .str.replace('.', '').str.replace('/', '_').str.replace(\"'\", '').str.replace(' ', '_')\n",
    "                                .str.split(\"__\").str[-1].str.split(\";\").str[-1])\n",
    "for n in range(2,7):\n",
    "    taxa_df.loc[taxa_df['Combo'].str.contains('uncultured_bacterium'),'Combo'] = taxa_df['Taxon'].astype(str)\\\n",
    "        .str.replace(';__','').str.replace('[','').str.replace(']','').str.replace('.','').str.replace('/','_')\\\n",
    "        .str.replace(\"'\",'').str.replace(' ','_').str.split(\"__\").str[-n].str.split(';').str[0]+'_unc_b'\n",
    "    taxa_df.loc[taxa_df['Combo'].str.contains('uncultured'),'Combo'] = taxa_df['Taxon'].astype(str)\\\n",
    "        .str.replace(';__','').str.replace('[','').str.replace(']','').str.replace('.','').str.replace('/','_')\\\n",
    "        .str.replace(\"'\",'').str.replace(' ','_').str.split(\"__\").str[-n].str.split(';').str[0]+'_unc'\n",
    "\n",
    "\n",
    "\n",
    "biom_df['#OTU ID'] = taxa_df['Combo']+'|'+taxa_df['Feature ID']\n",
    "taxa_df['Feature ID'] = biom_df['#OTU ID']\n",
    "taxa_df = taxa_df[['Feature ID', 'Taxon', 'Confidence']]\n",
    "\n",
    "#writing all files back\n",
    "biom_df.to_csv('Biom/feature-table.tsv', sep='\\t', index=False)\n",
    "taxa_df.to_csv('Taxa/taxonomy.tsv', sep='\\t', index=False)\n",
    "\n",
    "fasta_hash  = r\"Rep-seqs/dna-sequences.fasta\"\n",
    "fasta_combo = r\"Rep-seqs/dna-sequences.fa\"\n",
    "i=0\n",
    "with open(fasta_hash) as hashes, open(fasta_combo, 'w') as combo:\n",
    "    records = SeqIO.parse(fasta_hash, 'fasta')\n",
    "    for record in records:\n",
    "        record.id = biom_df['#OTU ID'][i]\n",
    "        record.description = ''\n",
    "        i+=1\n",
    "        SeqIO.write(record, combo, 'fasta')\n",
    "!rm $fasta_hash\n",
    "!mv $fasta_combo $fasta_hash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mImported Biom/feature-table.biom as BIOMV210Format to Data/tables/combo_table.qza\u001b[0m\n",
      "\u001b[32mImported Taxa/taxonomy.tsv as TSVTaxonomyDirectoryFormat to Data/combo_taxonomy.qza\u001b[0m\n",
      "\u001b[32mImported Rep-seqs/dna-sequences.fasta as DNASequencesDirectoryFormat to Data/combo_rep-seqs.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#creating new rep-seqs.qza, table.qza and taxonomy.qza with modified hashes\n",
    "!biom convert -i Biom/feature-table.tsv -o Biom/feature-table.biom --table-type=\"OTU table\" --to-hdf5\n",
    "!qiime tools import \\\n",
    "    --input-path Biom/feature-table.biom \\\n",
    "    --type 'FeatureTable[Frequency]' \\\n",
    "    --input-format BIOMV210Format \\\n",
    "    --output-path Data/tables/combo_table.qza\n",
    "!qiime tools import \\\n",
    "    --type 'FeatureData[Taxonomy]' \\\n",
    "    --input-path Taxa/taxonomy.tsv \\\n",
    "    --output-path Data/combo_taxonomy.qza\n",
    "!qiime tools import \\\n",
    "    --input-path $fasta_hash \\\n",
    "    --output-path Data/combo_rep-seqs.qza \\\n",
    "    --type 'FeatureData[Sequence]'\n",
    "\n",
    "!rm -r Biom Taxa Rep-seqs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate a tree for phylogenetic diversity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[AlignedSequence] to: Data/aligned-rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[AlignedSequence] to: Data/masked-aligned-rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Unrooted] to: Data/unrooted-tree.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Rooted] to: Data/rooted-tree.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "    --i-sequences Data/combo_rep-seqs.qza \\\n",
    "    --o-alignment Data/aligned-rep-seqs.qza \\\n",
    "    --o-masked-alignment Data/masked-aligned-rep-seqs.qza \\\n",
    "    --o-tree Data/unrooted-tree.qza \\\n",
    "    --o-rooted-tree Data/rooted-tree.qza \\\n",
    "    --p-n-threads 18"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filtration to remove low abundant features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/table-no-mit_chlor.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/tables/filtered-table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-features \\\n",
    "    --i-table Data/tables/combo_table.qza \\\n",
    "    --p-min-frequency 10 \\\n",
    "    --p-min-samples 5 \\\n",
    "    --o-filtered-table Data/tables/sample-contingency-filtered-table.qza\n",
    "\n",
    "!qiime taxa filter-table \\\n",
    "    --i-table Data/tables/sample-contingency-filtered-table.qza \\\n",
    "    --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "    --p-exclude mitochondria,chloroplast \\\n",
    "    --o-filtered-table Data/tables/table-no-mit_chlor.qza\n",
    "\n",
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/tables/table-no-mit_chlor.qza \\\n",
    "    --p-min-features 25 \\\n",
    "    --p-min-frequency 8000 \\\n",
    "    --o-filtered-table Data/tables/filtered-table.qza\n",
    "\n",
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/tables/filtered-table.qza \\\n",
    "    --m-metadata-file Data/tables/samples_to_exclude.tsv \\\n",
    "    --p-exclude-ids \\\n",
    "    --o-filtered-table Data/tables/filtered-table.qza\n",
    "\n",
    "!mkdir Results/tables\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/tables/filtered-table.qza \\\n",
    "    --o-visualization Results/tables/filtered-table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Excluding unassigned and assigned only to bacteria level features\n",
    "Extracted manually a taxonomy.tsv file from combo_taxonomy.qza and filtered it to leave only features to exclude."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/All_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/tables/All_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-features \\\n",
    "    --i-table Data/tables/filtered-table.qza \\\n",
    "    --m-metadata-file Data/tables/taxonomy_to_exclude.tsv \\\n",
    "    --p-exclude-ids \\\n",
    "    --o-filtered-table Data/tables/All_table.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/tables/All_table.qza \\\n",
    "    --o-visualization Results/tables/All_table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filtering by Treatments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Infected_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/tables/Infected_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/tables/All_table.qza \\\n",
    "    --m-metadata-file Metadata/metadata.tsv \\\n",
    "    --p-where \"Treatment='Infected'\" \\\n",
    "    --o-filtered-table Data/tables/Infected_table.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/tables/Infected_table.qza \\\n",
    "    --o-visualization Results/tables/Infected_table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filtering table by Niche and TimePoint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_table.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Filtering by Niche\n",
    "Niche_list = ['Root', 'Gall', 'J2', 'Soil']\n",
    "for Niche in Niche_list:\n",
    "    tabin  = 'Data/tables/Infected_table.qza'\n",
    "    tabout = 'Data/tables/%s_table.qza' % Niche\n",
    "    column = \"Niche='%s'\" % Niche     \n",
    "    \n",
    "    !qiime feature-table filter-samples \\\n",
    "        --i-table $tabin \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-where \"$column\" \\\n",
    "        --o-filtered-table $tabout \n",
    "\n",
    "    !qiime feature-table filter-features \\\n",
    "        --i-table $tabout \\\n",
    "        --p-min-frequency 3 \\\n",
    "        --p-min-samples 3 \\\n",
    "        --o-filtered-table $tabout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Gall_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-J2_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_1_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_1_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root-Soil_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-J2_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall-Soil_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2-Soil_6_table.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "Niche_list = ['Root', 'Gall', 'J2', 'Soil']\n",
    "for Niche_1, Niche_2 in itertools.combinations(Niche_list, 2):\n",
    "    tabin  = 'Data/tables/Infected_table.qza'\n",
    "    tabout = 'Data/tables/%s-%s_table.qza' % (Niche_1, Niche_2)\n",
    "    column = \"Niche IN ('%s','%s')\"        % (Niche_1, Niche_2)\n",
    "    \n",
    "    !qiime feature-table filter-samples \\\n",
    "        --i-table $tabin \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-where \"$column\" \\\n",
    "        --o-filtered-table $tabout \n",
    "    !qiime feature-table filter-features \\\n",
    "        --i-table $tabout \\\n",
    "        --p-min-frequency 3 \\\n",
    "        --p-min-samples 3 \\\n",
    "        --o-filtered-table $tabout\n",
    "    \n",
    "    tps = [1, 2, 3, 4, 5, 6]\n",
    "    if Niche_1 == 'Gall' or Niche_2 == 'Gall':\n",
    "        tps = [2, 3, 4, 5, 6]\n",
    "    if Niche_1 == 'J2' or Niche_2 == 'J2':\n",
    "        tps = [2, 3, 4, 6]\n",
    "        \n",
    "    for tp in tps:\n",
    "        tab_tp = 'Data/tables/%s-%s_%s_table.qza' % (Niche_1, Niche_2, tp)\n",
    "        col    = \"TimePoint='%s'\"                 % tp\n",
    "        !qiime feature-table filter-samples \\\n",
    "            --i-table $tabout \\\n",
    "            --m-metadata-file Metadata/metadata.tsv \\\n",
    "            --p-where \"$col\" \\\n",
    "            --o-filtered-table $tab_tp\n",
    "        !qiime feature-table filter-features \\\n",
    "            --i-table $tab_tp \\\n",
    "            --p-min-frequency 2 \\\n",
    "            --p-min-samples 2 \\\n",
    "            --o-filtered-table $tab_tp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_1-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_3-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_3-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_4-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_4-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_5-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_5-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_1-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_3-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_3-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_4-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_4-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_5-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_5-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_1-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_2-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_3-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_3-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_3-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_3-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_4-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_4-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_4-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_5-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_5-6_table.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "Niche_list = ['Root', 'Gall', 'J2', 'Soil']\n",
    "for Niche in Niche_list:\n",
    "    if Niche == 'Gall':\n",
    "        tps = [2, 3, 4, 5, 6]\n",
    "    if Niche == 'J2':\n",
    "        tps = [2, 3, 4, 6]\n",
    "    else:\n",
    "        tps = [1, 2, 3, 4, 5, 6]\n",
    "    for tp_1, tp_2 in itertools.combinations(tps, 2):\n",
    "        tabin  = 'Data/tables/%s_table.qza'       % Niche\n",
    "        tabout = 'Data/tables/%s_%s-%s_table.qza' % (Niche, tp_1, tp_2)\n",
    "        column = \"TimePoint IN ('%s','%s')\"       % (tp_1, tp_2)\n",
    "\n",
    "        !qiime feature-table filter-samples \\\n",
    "            --i-table $tabin \\\n",
    "            --m-metadata-file Metadata/metadata.tsv \\\n",
    "            --p-where \"$column\" \\\n",
    "            --o-filtered-table $tabout \n",
    "\n",
    "        !qiime feature-table filter-features \\\n",
    "            --i-table $tabout \\\n",
    "            --p-min-frequency 2 \\\n",
    "            --p-min-samples 2 \\\n",
    "            --o-filtered-table $tabout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Root-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Root-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Gall-J2_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-Gall-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-1-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-2-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-3-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-4-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-5-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-J2-Soil_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/tp-6-J2-Soil_table.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Filtering by TimePoint\n",
    "for TP in range(1,7):\n",
    "    tabin  = 'Data/tables/Infected_table.qza'\n",
    "    tabout = 'Data/tables/tp-%s_table.qza' % TP\n",
    "    column = \"TimePoint='%s'\" % TP  \n",
    "    !qiime feature-table filter-samples \\\n",
    "        --i-table $tabin \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-where \"$column\" \\\n",
    "        --o-filtered-table $tabout \n",
    "\n",
    "    !qiime feature-table filter-features \\\n",
    "        --i-table $tabout \\\n",
    "        --p-min-frequency 2 \\\n",
    "        --p-min-samples 2 \\\n",
    "        --o-filtered-table $tabout\n",
    "    \n",
    "import itertools\n",
    "Niche_list = ['Root', 'Gall', 'J2', 'Soil']\n",
    "for Niche_1, Niche_2 in itertools.combinations(Niche_list, 2):\n",
    "    for TP in range(1,7):\n",
    "        tabin  = 'Data/tables/%s-%s_table.qza'       % (Niche_1, Niche_2)\n",
    "        tabout = 'Data/tables/tp-%s-%s-%s_table.qza' % (TP, Niche_1, Niche_2)\n",
    "        column = \"TimePoint='%s'\"                    % TP  \n",
    "        !qiime feature-table filter-samples \\\n",
    "            --i-table $tabin \\\n",
    "            --m-metadata-file Metadata/metadata.tsv \\\n",
    "            --p-where \"$column\" \\\n",
    "            --o-filtered-table $tabout \n",
    "\n",
    "        !qiime feature-table filter-features \\\n",
    "            --i-table $tabout \\\n",
    "            --p-min-frequency 2 \\\n",
    "            --p-min-samples 2 \\\n",
    "            --o-filtered-table $tabout"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxa barplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Infected_Group_TP_grouped.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/Group_TP_taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Infected_Niche_grouped.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/Niche_taxabarplot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Taxa_barplots/Infected_taxabarplot.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "outdir = 'Results/Taxa_barplots'\n",
    "tabin  = 'Data/tables/Infected_table.qza'\n",
    "groups = ['Group_TP','Niche']#, 'Group_Treat']\n",
    "meta   = pd.read_csv('Metadata/metadata.tsv',sep='\\t',index_col=0)\n",
    "\n",
    "!mkdir -p $outdir\n",
    "\n",
    "for group in groups:\n",
    "#Collapse metadata\n",
    "    meta.index = meta[group]\n",
    "    meta = meta.drop_duplicates(subset=group)\n",
    "    meta.index.name = '#SampleID'\n",
    "    meta.to_csv('Metadata/%s_metadata.tsv'%group,sep='\\t')\n",
    "#Variables    \n",
    "    grouped = 'Data/tables/Infected_%s_grouped.qza' % group\n",
    "    taxabar = outdir + '/%s_taxabarplot.qzv'   % group\n",
    "    metadata= 'Metadata/%s_metadata.tsv'       % group\n",
    "#Group tables    \n",
    "    !qiime feature-table group \\\n",
    "        --i-table $tabin \\\n",
    "        --p-axis 'sample' \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --m-metadata-column $group \\\n",
    "        --p-mode 'mean-ceiling' \\\n",
    "        --o-grouped-table $grouped\n",
    "#Taxabarplots    \n",
    "    !qiime taxa barplot \\\n",
    "        --i-table $grouped \\\n",
    "        --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "        --m-metadata-file $metadata \\\n",
    "        --o-visualization $taxabar\n",
    "\n",
    "!qiime taxa barplot \\\n",
    "    --i-table Data/tables/Infected_table.qza \\\n",
    "    --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "    --m-metadata-file Metadata/metadata.tsv \\\n",
    "    --o-visualization Results/Taxa_barplots/Infected_taxabarplot.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alpha and beta diversity analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Results/Core-metrics_analysis’: File exists\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_alpha-rarefaction.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Alpha rarefaction plotting\n",
    "outdir = 'Results/Core-metrics_analysis'\n",
    "!mkdir $outdir\n",
    "variants = ['Infected'] #,\"All\",'Root-Gall']\n",
    "for variant in variants:\n",
    "    depth = 15000\n",
    "    alpha = outdir+'/%s_alpha-rarefaction.qzv' % variant\n",
    "    tabin = 'Data/tables/%s_table.qza' % variant\n",
    "    \n",
    "    !qiime diversity alpha-rarefaction \\\n",
    "        --i-table $tabin \\\n",
    "        --i-phylogeny Data/rooted-tree.qza \\\n",
    "        --p-max-depth $depth \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --o-visualization $alpha"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Core-metrics_analysis/All_Core-metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] % Properties('phylogenetic') to: Data/Core-metrics_analysis/All_Core-metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/All_Core-metrics/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/All_Core-metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/All_Core-metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Data/Core-metrics_analysis/All_Core-metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Data/Core-metrics_analysis/All_Core-metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Data/Core-metrics_analysis/All_Core-metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Data/Core-metrics_analysis/All_Core-metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/All_Core-metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/All_Core-metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/All_Core-metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/All_Core-metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/unweighted-unifrac-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/weighted-unifrac-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/jaccard-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/bray-curtis-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] % Properties('phylogenetic') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted-unifrac-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/weighted-unifrac-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/bray-curtis-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Core-metrics_analysis/Infected_Core-metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] % Properties('phylogenetic') to: Data/Core-metrics_analysis/Infected_Core-metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/Infected_Core-metrics/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/Infected_Core-metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Data/Core-metrics_analysis/Infected_Core-metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Data/Core-metrics_analysis/Infected_Core-metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Data/Core-metrics_analysis/Infected_Core-metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Data/Core-metrics_analysis/Infected_Core-metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Data/Core-metrics_analysis/Infected_Core-metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Infected_Core-metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Infected_Core-metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Infected_Core-metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Data/Core-metrics_analysis/Infected_Core-metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/unweighted-unifrac-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/weighted-unifrac-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/jaccard-emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/bray-curtis-emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# core-metrics-phylogenetic: Core diversity metrics (phylogenetic and non-phylogenetic)\n",
    "!mkdir Data/Core-metrics_analysis\n",
    "\n",
    "variants = ['Root-Gall', 'Infected']\n",
    "for variant in variants:\n",
    "    Data_core = 'Data/Core-metrics_analysis/%s_Core-metrics'    % variant\n",
    "    Rslt_core = 'Results/Core-metrics_analysis/%s_Core-metrics' % variant\n",
    "    tabin     = 'Data/tables/%s_table.qza' % variant\n",
    "    \n",
    "    !mkdir $Rslt_core\n",
    "    !qiime diversity core-metrics-phylogenetic \\\n",
    "        --i-phylogeny Data/rooted-tree.qza \\\n",
    "        --i-table $tabin \\\n",
    "        --p-sampling-depth 8828 \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-n-jobs 16 \\\n",
    "        --o-unweighted-unifrac-emperor $Rslt_core/unweighted-unifrac-emperor.qzv \\\n",
    "        --o-weighted-unifrac-emperor $Rslt_core/weighted-unifrac-emperor.qzv \\\n",
    "        --o-jaccard-emperor $Rslt_core/jaccard-emperor.qzv \\\n",
    "        --o-bray-curtis-emperor $Rslt_core/bray-curtis-emperor.qzv \\\n",
    "        --output-dir $Data_core"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/tables/All_relative_table.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/All_Core-metrics/unweighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/unweighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/All_Core-metrics/weighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/weighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/All_Core-metrics/jaccard_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/jaccard_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/All_Core-metrics/bray_curtis_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/bray_curtis_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/tables/Root-Gall_relative_table.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/weighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/weighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Root-Gall_Core-metrics/bray_curtis_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/bray_curtis_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/tables/Infected_relative_table.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Infected_Core-metrics/unweighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/unweighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Infected_Core-metrics/weighted_unifrac_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/weighted_unifrac_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Infected_Core-metrics/jaccard_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/jaccard_biplot.qzv\u001b[0m\n",
      "\u001b[32mSaved PCoAResults % Properties('biplot') to: Data/Core-metrics_analysis/Infected_Core-metrics/bray_curtis_biplot.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/bray_curtis_biplot.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Principal Coordinate Analysis Biplot\n",
    "variants = ['Root-Gall', 'Infected']\n",
    "for variant in variants:\n",
    "    Data_core = 'Data/Core-metrics_analysis/%s_Core-metrics'    % variant\n",
    "    Rslt_core = 'Results/Core-metrics_analysis/%s_Core-metrics' % variant\n",
    "    rel_tab   = 'Data/tables/%s_relative_table.qza'             % variant\n",
    "    tabin     = 'Data/tables/%s_table.qza'                      % variant\n",
    "    \n",
    "#Converting feature table [Frequency] to [Relative frequency]\n",
    "    !qiime feature-table relative-frequency \\\n",
    "        --i-table $tabin \\\n",
    "        --o-relative-frequency-table $rel_tab\n",
    "    \n",
    "#pcoa-biplot: Principal Coordinate Analysis Biplot\n",
    "    artifacts = ['unweighted_unifrac', 'weighted_unifrac', 'jaccard', 'bray_curtis']\n",
    "    for artifact in artifacts:\n",
    "        pcoa  = Data_core+'/%s_pcoa_results.qza' % artifact\n",
    "        biplot= Data_core+'/%s_biplot.qza'       % artifact\n",
    "        bi_qzv= Rslt_core+'/%s_biplot.qzv'       % artifact\n",
    "        \n",
    "        !qiime diversity pcoa-biplot \\\n",
    "            --i-pcoa $pcoa \\\n",
    "            --i-features $rel_tab \\\n",
    "            --o-biplot $biplot\n",
    "        !qiime emperor biplot \\\n",
    "            --i-biplot $biplot \\\n",
    "            --m-sample-metadata-file Metadata/metadata.tsv \\\n",
    "            --p-ignore-missing-samples \\\n",
    "            --p-number-of-features 5 \\\n",
    "            --o-visualization $bi_qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/shannon_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/faith_pd_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/evenness_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/observed_otus_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/shannon_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/faith_pd_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/evenness_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/observed_otus_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/shannon_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/faith_pd_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/evenness_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/observed_otus_vector_alpha-group-sign.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "variants = ['Root-Gall', 'Infected']\n",
    "for variant in variants:\n",
    "    core   = 'Data/Core-metrics_analysis/%s_Core-metrics'    % variant\n",
    "    outdir = 'Results/Core-metrics_analysis/%s_Core-metrics' % variant\n",
    "\n",
    "#Alpha diversity comparisons¶    \n",
    "    alpha_diversity = ['shannon', 'faith_pd', 'evenness', 'observed_otus']\n",
    "    for alpha in alpha_diversity:\n",
    "        vector  = core+'/%s_vector.qza'                      % alpha\n",
    "        ags_qzv = outdir+'/%s_vector_alpha-group-sign.qzv'   % alpha\n",
    "        \n",
    "        !qiime diversity alpha-group-significance \\\n",
    "            --i-alpha-diversity $vector \\\n",
    "            --m-metadata-file Metadata/metadata.tsv \\\n",
    "            --o-visualization $ags_qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/unweighted_unifrac_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/unweighted_unifrac_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/weighted_unifrac_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/weighted_unifrac_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/jaccard_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/jaccard_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/bray_curtis_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/All_Core-metrics/bray_curtis_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted_unifrac_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/unweighted_unifrac_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/weighted_unifrac_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/weighted_unifrac_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/jaccard_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/bray_curtis_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Root-Gall_Core-metrics/bray_curtis_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/unweighted_unifrac_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/unweighted_unifrac_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/weighted_unifrac_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/weighted_unifrac_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/jaccard_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/jaccard_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/bray_curtis_Group_TP-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis/Infected_Core-metrics/bray_curtis_Group_Treat-dist-matrix_beta-group-sign-pairwise.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Beta diversity comparisons¶\n",
    "variants = ['Root-Gall', 'Infected']\n",
    "for variant in variants:\n",
    "    core   = 'Data/Core-metrics_analysis/%s_Core-metrics'    % variant\n",
    "    outdir = 'Results/Core-metrics_analysis/%s_Core-metrics' % variant\n",
    "    beta_diversity = ['unweighted_unifrac', 'weighted_unifrac', 'jaccard', 'bray_curtis']\n",
    "    for beta in beta_diversity:\n",
    "        distance= core+'/%s_distance_matrix.qza' % beta\n",
    "        columns = ['Group_TP', 'Group_Treat']\n",
    "        for column in columns:\n",
    "            bgs_qzv = outdir+'/%s_%s-dist-matrix_beta-group-sign-pairwise.qzv' % (beta, column)\n",
    "            \n",
    "            !qiime diversity beta-group-significance \\\n",
    "                --i-distance-matrix $distance \\\n",
    "                --m-metadata-file Metadata/metadata.tsv \\\n",
    "                --m-metadata-column $column \\\n",
    "                --p-pairwise \\\n",
    "                --o-visualization $bgs_qzv         "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Core features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_2_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_2_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_2_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_2_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_3_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_3_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_3_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_3_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_4_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_4_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_4_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_4_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_6_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/S_6_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_6_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/S_6_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_2_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_2_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_2_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_2_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_3_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_3_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_3_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_3_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_4_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_4_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_4_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_4_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_6_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/R_6_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_6_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/R_6_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_2_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_2_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_2_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_2_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_3_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_3_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_3_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_3_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_4_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_4_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_4_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_4_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_6_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/G_6_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_6_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/G_6_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_2_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_2_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_2_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_2_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_3_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_3_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_3_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_3_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_4_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_4_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_4_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_4_core.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_6_table.qza\u001b[0m\n",
      "mkdir: missing operand\n",
      "Try 'mkdir --help' for more information.\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J_6_7-lev_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_6_7-lev_core.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core_Features/J_6_core.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#By Niches and TimePoints\n",
    "outdir= 'Results/Core_Features'\n",
    "groups = ['S_2','S_3','S_4','S_6',\n",
    "          'R_2','R_3','R_4','R_6',\n",
    "          'G_2','G_3','G_4','G_6',\n",
    "          'J_2','J_3','J_4','J_6']\n",
    "\n",
    "!mkdir $outdir\n",
    "for group in groups:\n",
    "    tabin  = 'Data/tables/Infected_table.qza'\n",
    "    column = \"Group_TP='%s'\"            % group\n",
    "    tabout = 'Data/tables/%s_table.qza' % group\n",
    "\n",
    "    !qiime feature-table filter-samples \\\n",
    "        --i-table $tabin \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-where \"$column\" \\\n",
    "        --o-filtered-table $tabout\n",
    "     \n",
    "    table_7= 'Data/tables/%s_7-lev_table.qza'  % group\n",
    "    core_7 = outdir+'/%s_7-lev_core.qzv'       % group\n",
    "    coreqzv= outdir+'/%s_core.qzv'             % group\n",
    "\n",
    "    !qiime taxa collapse \\\n",
    "        --i-table $tabout \\\n",
    "        --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "        --p-level 7 \\\n",
    "        --o-collapsed-table $table_7\n",
    "    !qiime feature-table core-features \\\n",
    "        --i-table $table_7 \\\n",
    "        --o-visualization $core_7\n",
    "    !qiime feature-table core-features \\\n",
    "        --i-table $tabout \\\n",
    "        --o-visualization $coreqzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Core-features_tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# By Niches and TimePoints separately \n",
    "import pandas as pd\n",
    "outdir= 'Results/Core_Features'\n",
    "groups = ['S_2','S_3','S_4','S_6',\n",
    "          'R_2','R_3','R_4','R_6',\n",
    "          'G_2','G_3','G_4','G_6',\n",
    "          'J_2','J_3','J_4','J_6']\n",
    "\n",
    "for group in groups:\n",
    "    temp = 'Data/Core_%s_temp' % group\n",
    "    coreqzv = 'Results/Core_Features/%s_core.qzv' % group\n",
    "    !mkdir $temp\n",
    "    !qiime tools export \\\n",
    "        --input-path $coreqzv \\\n",
    "        --output-path $temp\n",
    "\n",
    "    temp_7 = 'Data/Core_%s_temp_7' % group\n",
    "    core_7 = 'Results/Core_Features/%s_7-lev_core.qzv' % group\n",
    "    !mkdir $temp_7\n",
    "    !qiime tools export \\\n",
    "        --input-path $core_7 \\\n",
    "        --output-path $temp_7\n",
    "\n",
    "Root, Gall, Soil, J2 = pd.DataFrame(), pd.DataFrame(), pd.DataFrame(), pd.DataFrame()\n",
    "for group in groups:\n",
    "    temp = 'Data/Core_%s_temp' % group\n",
    "    core_tsv= temp+'/core-features-1.000.tsv' \n",
    "    core_pd = pd.read_csv(core_tsv, sep='\\t')\n",
    "    core_pd = core_pd.filter(['Feature ID'], axis = 1)\n",
    "    core_pd.columns = [group]\n",
    "    if 'S' in group: Soil = pd.concat([Soil, core_pd], axis = 1)\n",
    "    if 'R' in group: Root = pd.concat([Root, core_pd], axis = 1)\n",
    "    if 'G' in group: Gall = pd.concat([Gall, core_pd], axis = 1)\n",
    "    if 'J' in group: J2   = pd.concat([J2, core_pd], axis = 1)\n",
    "    !rm -r $temp\n",
    "        \n",
    "Soil.to_csv('Results/Core_Features/Soil_tp.tsv', sep='\\t')\n",
    "Root.to_csv('Results/Core_Features/Root_tp.tsv', sep='\\t')\n",
    "Gall.to_csv('Results/Core_Features/Gall_tp.tsv', sep='\\t')\n",
    "J2.to_csv('Results/Core_Features/J2_tp.tsv', sep='\\t')\n",
    "\n",
    "Root, Gall, Soil, J2 = pd.DataFrame(), pd.DataFrame(), pd.DataFrame(), pd.DataFrame()\n",
    "for group in groups:\n",
    "    temp_7 = 'Data/Core_%s_temp_7' % group\n",
    "    core_tsv= temp_7+'/core-features-1.000.tsv' \n",
    "    core_pd = pd.read_csv(core_tsv, sep='\\t')\n",
    "    core_pd = core_pd.filter(['Feature ID'], axis = 1)\n",
    "    core_pd.columns = [group]\n",
    "    if 'S' in group: Soil = pd.concat([Soil, core_pd], axis = 1)\n",
    "    if 'R' in group: Root = pd.concat([Root, core_pd], axis = 1)\n",
    "    if 'G' in group: Gall = pd.concat([Gall, core_pd], axis = 1)\n",
    "    if 'J' in group: J2   = pd.concat([J2, core_pd], axis = 1)\n",
    "    !rm -r $temp_7\n",
    "        \n",
    "Soil.to_csv('Results/Core_Features/Soil_tp_lev-7.tsv', sep='\\t')\n",
    "Root.to_csv('Results/Core_Features/Root_tp_lev-7.tsv', sep='\\t')\n",
    "Gall.to_csv('Results/Core_Features/Gall_tp_lev-7.tsv', sep='\\t')\n",
    "J2.to_csv('Results/Core_Features/J2_tp_lev-7.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LONGITUDINAL"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pairwise differences & distances comparisons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_shannon_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_faith_pd_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_evenness_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-differences/Infected_observed_otus_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_shannon_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_faith_pd_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_evenness_pairwise-differences_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-differences/Root-Gall_observed_otus_pairwise-differences_parametric_4-6.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Differences\n",
    "import itertools\n",
    "timepoints = [2,3,4,6]\n",
    "variants = ['Infected', 'Root-Gall']\n",
    "for variant in variants:\n",
    "    core_dt = 'Data/Core-metrics_analysis/%s_Core-metrics' % variant\n",
    "    core_rs = 'Results/Longitudinal/%s'                    % variant\n",
    "    tabin = 'Data/tables/%s_table.qza'                     % variant\n",
    "    \n",
    "    !mkdir -p $core_dt $core_rs/Pairwise-differences\n",
    "    for state, state_2 in itertools.combinations(timepoints, 2):\n",
    "        alpha_diversity = ['shannon', 'faith_pd', 'evenness', 'observed_otus']\n",
    "        for alpha in alpha_diversity:\n",
    "            vector=core_dt+'/%s_vector.qza' % alpha\n",
    "            param =core_rs+'/Pairwise-differences/%s_%s_pairwise-differences_parametric_%s-%s.qzv'   %(variant,alpha,state,state_2)\n",
    "            no_par=core_rs+'/Pairwise-differences/%s_%s_pairwise-differences_no-parametric_%s-%s.qzv'%(variant,alpha,state,state_2)\n",
    "            if alpha == 'evenness':\n",
    "                metric = 'pielou_e'\n",
    "            else:\n",
    "                metric = alpha\n",
    "            !qiime longitudinal pairwise-differences \\\n",
    "                --m-metadata-file Metadata/metadata.tsv \\\n",
    "                --m-metadata-file $vector \\\n",
    "                --p-metric $metric \\\n",
    "                --p-group-column Niche \\\n",
    "                --p-state-column TimePoint \\\n",
    "                --p-state-1 $state \\\n",
    "                --p-state-2 $state_2 \\\n",
    "                --p-individual-id-column Location \\\n",
    "                --p-replicate-handling random \\\n",
    "                --p-no-parametric \\\n",
    "                --o-visualization $no_par\n",
    "            !qiime longitudinal pairwise-differences \\\n",
    "                --m-metadata-file Metadata/metadata.tsv \\\n",
    "                --m-metadata-file $vector \\\n",
    "                --p-metric $metric \\\n",
    "                --p-group-column Niche \\\n",
    "                --p-state-column TimePoint \\\n",
    "                --p-state-1 $state \\\n",
    "                --p-state-2 $state_2 \\\n",
    "                --p-individual-id-column Location \\\n",
    "                --p-replicate-handling random \\\n",
    "                --p-parametric \\\n",
    "                --o-visualization $param   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Results/Longitudinal/Infected/Pairwise-distances’: File exists\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_unweighted_unifrac_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_weighted_unifrac_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_jaccard_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Pairwise-distances/Infected_bray_curtis_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "mkdir: cannot create directory ‘Results/Longitudinal/Root-Gall/Pairwise-distances’: File exists\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_unweighted_unifrac_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_weighted_unifrac_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_jaccard_pairwise-distances_parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_no-parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_parametric_2-3.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_no-parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_parametric_2-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_no-parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_parametric_2-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_no-parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_parametric_3-4.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_no-parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_parametric_3-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_no-parametric_4-6.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Pairwise-distances/Root-Gall_bray_curtis_pairwise-distances_parametric_4-6.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "#Distances\n",
    "import itertools\n",
    "timepoints = [2,3,4,6]\n",
    "variants = ['Infected', 'Root-Gall']\n",
    "for variant in variants:\n",
    "    core_dt = 'Data/Core-metrics_analysis/%s_Core-metrics' % variant\n",
    "    core_rs = 'Results/Longitudinal/%s'                    % variant\n",
    "    tabin = 'Data/tables/%s_table.qza'                     % variant\n",
    "    \n",
    "    !mkdir $core_rs/Pairwise-distances\n",
    "    beta_diversity = ['unweighted_unifrac', 'weighted_unifrac', 'jaccard', 'bray_curtis']\n",
    "    for beta in beta_diversity:\n",
    "        for state, state_2 in itertools.combinations(timepoints, 2):\n",
    "            distance= core_dt+'/%s_distance_matrix.qza' % beta\n",
    "            output  = core_rs+'/Pairwise-distances/%s_%s_pairwise-distances_no-parametric_%s-%s.qzv' % (variant, beta, state, state_2)\n",
    "            param   = core_rs+'/Pairwise-distances/%s_%s_pairwise-distances_parametric_%s-%s.qzv'    % (variant, beta, state, state_2)\n",
    "            \n",
    "            !qiime longitudinal pairwise-distances \\\n",
    "                --i-distance-matrix $distance \\\n",
    "                --m-metadata-file Metadata/metadata.tsv \\\n",
    "                --p-group-column Niche \\\n",
    "                --p-state-column TimePoint \\\n",
    "                --p-state-1 $state \\\n",
    "                --p-state-2 $state_2 \\\n",
    "                --p-individual-id-column Location \\\n",
    "                --p-replicate-handling random \\\n",
    "                --o-visualization $output\n",
    "            !qiime longitudinal pairwise-distances \\\n",
    "                --i-distance-matrix $distance \\\n",
    "                --m-metadata-file Metadata/metadata.tsv \\\n",
    "                --p-group-column Niche \\\n",
    "                --p-state-column TimePoint \\\n",
    "                --p-state-1 $state \\\n",
    "                --p-state-2 $state_2 \\\n",
    "                --p-individual-id-column Location \\\n",
    "                --p-replicate-handling random \\\n",
    "                --p-parametric \\\n",
    "                --o-visualization $param"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Feature volatility analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Gall_table_7_all_tp.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/Gall_Taxa_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/Gall_Taxa_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/Gall_Taxa_Feature_Volatility/Gall_Taxa_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/Gall_Taxa_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/Gall_Taxa_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Root_table_7_all_tp.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/Root_Taxa_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/Root_Taxa_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/Root_Taxa_Feature_Volatility/Root_Taxa_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/Root_Taxa_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/Root_Taxa_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/J2_table_7_all_tp.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/J2_Taxa_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/J2_Taxa_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/J2_Taxa_Feature_Volatility/J2_Taxa_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/J2_Taxa_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/J2_Taxa_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/tables/Soil_table_7_all_tp.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/Soil_Taxa_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/Soil_Taxa_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/Soil_Taxa_Feature_Volatility/Soil_Taxa_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/Soil_Taxa_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/Soil_Taxa_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/Gall_ASV_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/Gall_ASV_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/Gall_ASV_Feature_Volatility/Gall_ASV_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/Gall_ASV_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/Gall_ASV_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/Root_ASV_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/Root_ASV_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/Root_ASV_Feature_Volatility/Root_ASV_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/Root_ASV_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/Root_ASV_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/J2_ASV_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/J2_ASV_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/J2_ASV_Feature_Volatility/J2_ASV_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/J2_ASV_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/J2_ASV_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/By_Niches/Soil_ASV_Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/By_Niches/Soil_ASV_Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/By_Niches/Soil_ASV_Feature_Volatility/Soil_ASV_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/By_Niches/Soil_ASV_Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/By_Niches/Soil_ASV_Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "rm: cannot remove 'Data/Longitudinal/By_Niches': Is a directory\n"
     ]
    }
   ],
   "source": [
    "#By_Niches\n",
    "niches = ['Gall','Root','J2','Soil']\n",
    "levels = ['Taxa','ASV']\n",
    "for level in levels:\n",
    "    for niche in niches:\n",
    "        out_rs = 'Results/Longitudinal/By_Niches/%s_%s_Feature_Volatility' %(niche,level)\n",
    "        out_dt = 'Data/Longitudinal/By_Niches/%s_%s_Feature_Volatility'    %(niche,level)\n",
    "        volplot= out_rs + '/%s_%s_volatility-plot.qzv'                     %(niche,level)\n",
    "        tabin  = 'Data/tables/%s_table.qza'                                %niche\n",
    "        tabin7 = 'Data/tables/%s_table_7_all_tp.qza'                       %niche\n",
    "        if level == 'Taxa':\n",
    "            table = tabin7\n",
    "            !qiime taxa collapse \\\n",
    "                --i-table $tabin \\\n",
    "                --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "                --p-level 7 \\\n",
    "                --o-collapsed-table $tabin7\n",
    "        else:\n",
    "            table = tabin\n",
    "        !mkdir -p $out_rs \n",
    "        !qiime longitudinal feature-volatility \\\n",
    "            --i-table $table \\\n",
    "            --m-metadata-file Metadata/metadata.tsv \\\n",
    "            --p-state-column TimePoint \\\n",
    "            --p-individual-id-column Location \\\n",
    "            --output-dir $out_dt \\\n",
    "            --p-n-jobs 16 \\\n",
    "            --p-feature-count 'all' \\\n",
    "            --o-volatility-plot $volplot\n",
    "#!rm Data/Longitudinal/By_Niches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/Infected/Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/Infected/Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Feature_Volatility/Infected_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/Infected/Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/Infected/Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Longitudinal/Infected_7_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/Infected/Feature_Volatility_lev-7/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/Infected/Feature_Volatility_lev-7/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Infected/Feature_Volatility_lev-7/Infected_volatility-plot_7-lev.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/Infected/Feature_Volatility_lev-7/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/Infected/Feature_Volatility_lev-7/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/Root-Gall/Feature_Volatility/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/Root-Gall/Feature_Volatility/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Feature_Volatility/Root-Gall_volatility-plot.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/Root-Gall/Feature_Volatility/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/Root-Gall/Feature_Volatility/sample_estimator.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/Longitudinal/Root-Gall_7_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[RelativeFrequency] to: Data/Longitudinal/Root-Gall/Feature_Volatility_lev-7/filtered_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Importance] to: Data/Longitudinal/Root-Gall/Feature_Volatility_lev-7/feature_importance.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Longitudinal/Root-Gall/Feature_Volatility_lev-7/Root-Gall_volatility-plot_7-lev.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/Longitudinal/Root-Gall/Feature_Volatility_lev-7/accuracy_results.qzv\u001b[0m\n",
      "\u001b[32mSaved SampleEstimator[Regressor] to: Data/Longitudinal/Root-Gall/Feature_Volatility_lev-7/sample_estimator.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "variants = ['Infected', 'Root-Gall']\n",
    "for variant in variants:\n",
    "    tabin  = 'Data/tables/%s_table.qza'                   % variant\n",
    "    out_rs = 'Results/Longitudinal/%s/Feature_Volatility' % variant\n",
    "    out_dt = 'Data/Longitudinal/%s/Feature_Volatility'    % variant\n",
    "    volplot= out_rs + '/%s_volatility-plot.qzv'           % variant\n",
    "    \n",
    "    !mkdir -p $out_rs \n",
    "    !qiime longitudinal feature-volatility \\\n",
    "        --i-table $tabin \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-state-column TimePoint \\\n",
    "        --p-individual-id-column Location \\\n",
    "        --output-dir $out_dt \\\n",
    "        --p-n-jobs 16 \\\n",
    "        --p-feature-count 'all' \\\n",
    "        --o-volatility-plot $volplot\n",
    "    \n",
    "   \n",
    "    out_rs_level = 'Results/Longitudinal/%s/Feature_Volatility_lev-7' % variant\n",
    "    out_dt_level = 'Data/Longitudinal/%s/Feature_Volatility_lev-7'    % variant\n",
    "    tabout_level = 'Data/Longitudinal/%s_7_table.qza'                 % variant\n",
    "    volplot_level= out_rs_level + '/%s_volatility-plot_7-lev.qzv'     % variant\n",
    "    \n",
    "    !mkdir -p $out_rs_level  \n",
    "    !qiime taxa collapse \\\n",
    "        --i-table $tabin \\\n",
    "        --i-taxonomy Data/combo_taxonomy.qza \\\n",
    "        --p-level 7 \\\n",
    "        --o-collapsed-table $tabout_level\n",
    "    !qiime longitudinal feature-volatility \\\n",
    "        --i-table $tabout_level \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-state-column TimePoint \\\n",
    "        --p-individual-id-column Location \\\n",
    "        --p-n-jobs 16 \\\n",
    "        --p-feature-count 'all' \\\n",
    "        --output-dir $out_dt_level \\\n",
    "        --o-volatility-plot $volplot_level\n",
    "#!rm Data/Longitudinal"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 97% OTU Clustering and core metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘Data/97_OTU’: File exists\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/table_97-OTU.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: Data/97_OTU/rep-seqs_97.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/97_OTU/table_97-OTU.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!mkdir Data/97_OTU\n",
    "!qiime vsearch cluster-features-de-novo \\\n",
    "    --i-table Data/tables/table.qza \\\n",
    "    --i-sequences Data/rep-seqs.qza \\\n",
    "    --p-perc-identity 0.97 \\\n",
    "    --o-clustered-table Data/97_OTU/table_97-OTU.qza \\\n",
    "    --o-clustered-sequences Data/97_OTU/rep-seqs_97.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/97_OTU/table_97-OTU.qza  \\\n",
    "    --o-visualization Data/97_OTU/table_97-OTU.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/97_OTU/filtered-table.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/filtered-table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/97_OTU/filtered-table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-features \\\n",
    "    --i-table Data/97_OTU/table_97-OTU.qza \\\n",
    "    --p-min-frequency 10 \\\n",
    "    --p-min-samples 5 \\\n",
    "    --o-filtered-table Data/97_OTU/filtered-table.qza\n",
    "\n",
    "!qiime taxa filter-table \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --i-taxonomy Data/taxonomy.qza \\\n",
    "    --p-exclude mitochondria,chloroplast \\\n",
    "    --o-filtered-table Data/97_OTU/filtered-table.qza\n",
    "\n",
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --p-min-features 25 \\\n",
    "    --p-min-frequency 8000 \\\n",
    "    --o-filtered-table Data/97_OTU/filtered-table.qza\n",
    "\n",
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --m-metadata-file Data/tables/samples_to_exclude.tsv \\\n",
    "    --p-exclude-ids \\\n",
    "    --o-filtered-table Data/97_OTU/filtered-table.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --o-visualization Data/97_OTU/filtered-table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv\n",
    "\n",
    "!qiime taxa filter-table \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --i-taxonomy Data/taxonomy.qza \\\n",
    "    --p-include D_1__ \\\n",
    "    --o-filtered-table Data/97_OTU/filtered-table.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --o-visualization Data/97_OTU/filtered-table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/Infected_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/97_OTU/Infected_table.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Data/97_OTU/Root-Gall_table.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Data/97_OTU/Root-Gall_table.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/97_OTU/filtered-table.qza \\\n",
    "    --m-metadata-file Metadata/metadata.tsv \\\n",
    "    --p-where \"Treatment='Infected'\" \\\n",
    "    --o-filtered-table Data/97_OTU/Infected_table.qza\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/97_OTU/Infected_table.qza \\\n",
    "    --o-visualization Data/97_OTU/Infected_table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv\n",
    "    \n",
    "!qiime feature-table filter-samples \\\n",
    "    --i-table Data/97_OTU/Infected_table.qza \\\n",
    "    --m-metadata-file Metadata/metadata.tsv \\\n",
    "    --p-where \"Niche IN ('Root','Gall')\" \\\n",
    "    --o-filtered-table Data/97_OTU/Root-Gall_table.qza\n",
    "!qiime feature-table summarize \\\n",
    "    --i-table Data/97_OTU/Root-Gall_table.qza \\\n",
    "    --o-visualization Data/97_OTU/Root-Gall_table.qzv \\\n",
    "    --m-sample-metadata-file Metadata/metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureData[AlignedSequence] to: Data/97_OTU/aligned-rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[AlignedSequence] to: Data/97_OTU/masked-aligned-rep-seqs.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Unrooted] to: Data/97_OTU/unrooted-tree.qza\u001b[0m\n",
      "\u001b[32mSaved Phylogeny[Rooted] to: Data/97_OTU/rooted-tree_97-OTU.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime phylogeny align-to-tree-mafft-fasttree \\\n",
    "    --i-sequences Data/97_OTU/rep-seqs_97.qza \\\n",
    "    --o-alignment Data/97_OTU/aligned-rep-seqs.qza \\\n",
    "    --o-masked-alignment Data/97_OTU/masked-aligned-rep-seqs.qza \\\n",
    "    --o-tree Data/97_OTU/unrooted-tree.qza \\\n",
    "    --o-rooted-tree Data/97_OTU/rooted-tree_97-OTU.qza \\\n",
    "    --p-n-threads 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] % Properties('phylogenetic') to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/unweighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/weighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/bray_curtis_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] % Properties('phylogenetic') to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/faith_pd_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/observed_otus_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/evenness_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/unweighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix % Properties('phylogenetic') to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/weighted_unifrac_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/unweighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/weighted_unifrac_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/unweighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/weighted_unifrac_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/bray_curtis_emperor.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!mkdir Results/Core-metrics_analysis_97-OTU\n",
    "\n",
    "variants = ['Root-Gall', 'Infected']\n",
    "for variant in variants:\n",
    "    Rslt_core = 'Results/Core-metrics_analysis_97-OTU/%s_Core-metrics' % variant\n",
    "    tabin     = 'Data/97_OTU/%s_table.qza'    % variant\n",
    "\n",
    "    !qiime diversity core-metrics-phylogenetic \\\n",
    "        --i-phylogeny Data/97_OTU/rooted-tree_97-OTU.qza \\\n",
    "        --i-table $tabin \\\n",
    "        --p-sampling-depth 8867 \\\n",
    "        --m-metadata-file Metadata/metadata.tsv \\\n",
    "        --p-n-jobs 4 \\\n",
    "        --output-dir $Rslt_core"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/shannon_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/faith_pd_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/evenness_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Root-Gall_Core-metrics/observed_otus_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/shannon_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/faith_pd_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/evenness_vector_alpha-group-sign.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: Results/Core-metrics_analysis_97-OTU/Infected_Core-metrics/observed_otus_vector_alpha-group-sign.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "variants = ['Root-Gall', 'Infected']\n",
    "for variant in variants:\n",
    "    core   = 'Results/Core-metrics_analysis_97-OTU/%s_Core-metrics' % variant\n",
    "#Alpha diversity comparisons¶    \n",
    "    alpha_diversity = ['shannon', 'faith_pd', 'evenness', 'observed_otus']\n",
    "    for alpha in alpha_diversity:\n",
    "        vector  = core+'/%s_vector.qza'                    % alpha\n",
    "        ags_qzv = core+'/%s_vector_alpha-group-sign.qzv'   % alpha\n",
    "        \n",
    "        !qiime diversity alpha-group-significance \\\n",
    "            --i-alpha-diversity $vector \\\n",
    "            --m-metadata-file Metadata/metadata.tsv \\\n",
    "            --o-visualization $ags_qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
